tool/plugin/lib/plugin.dart

```diff
@@ -638,121 +638,161 @@ class BuildCommand extends ProductCommand {
-        source = processedFile.readAsStringSync();
-        files[processedFile] = source;
-        source = source.replaceAll(
-          'import com.android.tools.idea.npw.model.ProjectSyncInvoker',
-          'import com.android.tools.idea.npw.model.NewModuleModel',
-        );
-        source = source.replaceAll(
-          'createStep(@NotNull Project model, @NotNull ProjectSyncInvoker invoker, String parent)',
-          'createStep(@NotNull NewModuleModel model)',
-        );
-        source = source.replaceAll(
-          'FlutterProjectModel model(@NotNull Project project,',
-          'FlutterProjectModel model(@NotNull NewModuleModel project,',
-        );
-        source = source.replaceAll(
-          'mySharedModel.getValue().project().setValue(project);',
-          'mySharedModel.getValue().project().setValue(project.getProject().getValue());',
-        );
-        processedFile.writeAsStringSync(source);
+        if (!spec.version.startsWith('4.1')) {
 
-      } else if (spec.version.startsWith('4.0')) {
+          processedFile = File(
+              'src/io/flutter/utils/AndroidUtils.java');
+          source = processedFile.readAsStringSync();
+          files[processedFile] = source;
+          source = source.replaceAll(
+            'import com.android.tools.idea.gradle.dsl.model.BuildModelContext;',
+            'import com.android.tools.idea.gradle.dsl.parser.BuildModelContext;',
+          );
+          processedFile.writeAsStringSync(source);
 
-        processedFile = File(
-            'flutter-studio/src/io/flutter/project/FlutterProjectSystem.java');
-        source = processedFile.readAsStringSync();
-        files[processedFile] = source;
-        source = source.replaceAll(
-          'gradleProjectSystem.getAndroidFacetsWithPackageName(project, packageName, scope)',
-          'Collections.emptyList()',
-        );
-        processedFile.writeAsStringSync(source);
+          processedFile = File(
+              'flutter-studio/src/io/flutter/actions/FlutterShowStructureSettingsAction.java');
+          source = processedFile.readAsStringSync();
+          files[processedFile] = source;
+          source = source.replaceAll(
+            'import com.android.tools.idea.gradle.structure.actions.AndroidShowStructureSettingsAction;',
+            'import com.android.tools.idea.gradle.actions.AndroidShowStructureSettingsAction;',
+          );
+          processedFile.writeAsStringSync(source);
+
+          processedFile = File(
+              'flutter-studio/src/io/flutter/actions/OpenAndroidModule.java');
+          source = processedFile.readAsStringSync();
+          files[processedFile] = source;
+          source = source.replaceAll(
+            'findGradleTarget', // 3 matches
+            'findImportTarget',
+          );
+          processedFile.writeAsStringSync(source);
+        }
+      } catch (ex) {
+        // Restore sources.
```

Comment:
Do we want this restore code to happen on an exception, or in a `finally`?

Reply:
It happens in a finally after the compiler is finished.